modules/bcftools/concat/main.nf

```diff
@@ -0,0 +1,39 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process BCFTOOLS_CONCAT {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.11" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bcftools:1.11--h7c999a4_0"
+    } else {
+        container "quay.io/biocontainers/bcftools:1.11--h7c999a4_0"
+    }
+
+    input:
+    tuple val(meta), path(vcfs)
+
+    output:
+    tuple val(meta), path("*.gz"), emit: vcf
+    path  "*.version.txt"        , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    prefix       = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    """
+    bcftools concat \\
+        --output ${prefix}.vcf.gz \\
+        $options.args \\
+        *.vcf.gz
```

Comment:
```suggestion
        ${vcfs}
```

For safety you could so something like `${vcfs}.join('  ')` or something, just a suggestion though. 

I guess you can't guarantee that a user supplies ungzipped VCFs, so safer to supply just whatever the user inputs

Reply:
Thanks for the pointer on this @jfy133  - makes sense 👍

Reply:
I'll be adding a few more modules later today before I open up this PR for a more general review - shall I ping you then?

Reply:
Sure! Although not sure if I can review today. Just make sure each module is a separate PR as per guidelines!